src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/Search/Expressions/IncludeRewriterTests.cs

```diff
@@ -0,0 +1,225 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions;
+using Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors;
+using Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors.QueryGenerators;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.SqlServer.UnitTests.Features.Search.Expressions
+{
+    public class IncludeRewriterTests
+    {
+        private readonly SearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private IReadOnlyList<string> _includeTargetTypes = new List<string>() { "MedicationRequest" };
+
+        public IncludeRewriterTests()
+        {
+            ModelInfoProvider.SetProvider(new VersionSpecificModelInfoProvider());
+            _searchParameterDefinitionManager = new SearchParameterDefinitionManager(ModelInfoProvider.Instance);
+            _searchParameterDefinitionManager.Start();
+        }
+
+        [Fact]
+        public void GivenASqlRootExpressionWithIncludes_WhenVisitedByIncludeRewriter_OrderIterateExpressionsAfterOtherSearchParametersAndAfterIncludeExpressionsTheyAreIteratingOver()
+        {
+            // Order the following query:
+            // [base]/MedicationDispense?_include:iterate=Patient:general-practitioner&_include:iterate=MedicationRequest:patient&_include=MedicationDispense:prescription&_id=smart-MedicationDispense-567
+
+            var refSearchParameter = _searchParameterDefinitionManager.GetSearchParameter("Patient", "general-practitioner");
```

Comment:
The challenge is coming up with the URI.
For example, in Patient:general-practitioner, the URI is http://hl7.org/fhir/SearchParameter/Patient-general-practitioner.
If indeed it follows the same pattern for all the other resources and references then there shouldn't be a problem. 

I'll look into it.

Reply:
I do agree in principle, however, by only linking to Stu3 it misses R4 and R5 Search parameter definitions which may be different. Keeping the DataLayer projects free of version specific references has been very helpful with maintaining multiple versions of FHIR. If we want the full coverage in all versions of FHIR would it make sense to move these to https://github.com/microsoft/fhir-server/tree/master/test/Microsoft.Health.Fhir.Shared.Tests.Integration? 
I know its a SQL Specific rewriter, but there isn't a SQL unit test project that offers this coverage, thoughts? //cc @johnstairs

Reply:
The way I see it, the R4 and R5 parameter definitions are not needed for this uni tests since we generated the relevant test cases for the unit tests using Stu3 parameters. We are not checking all the possible parameters and their version fit, we are checking a sorting algorithm... we need a few parameters (not all of them)